extras/wait-for-bricks.sh

```diff
@@ -92,34 +48,36 @@ function volumes_waiting_for_bricks_up()
         local vol_down_info
         while read -r volname
         do
-                if [[ "Started" == "$(volinfo_field "$volname" 'statusStr')" ]]
+                read -r total_brick_count
+                brick_count="$(volume_online_brick_count "$volname" "$total_brick_count")"
+                if [[ $brick_count -ne $total_brick_count ]]
                 then
-                        if [[ "Y" == "$(is_enabled "$(volinfo_option "$volname" features.shard)")" ]]
+                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
+                        if [[ -z "$wait_info" ]]
                         then
-                                brick_count="$(volume_online_brick_count "$volname")"
-                                total_brick_count=$(volinfo_field "$volname" 'brickCount')
-                                if [[ $brick_count -ne $total_brick_count ]]
-                                then
-                                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
-                                        if [[ -z "$wait_info" ]]
-                                        then
-                                                wait_info="$vol_down_info"
-                                        else
-                                                wait_info="$wait_info, $vol_down_info"
-                                        fi
-                                fi
+                                wait_info="$vol_down_info"
+                        else
+                                wait_info="$wait_info, $vol_down_info"
                         fi
                 fi
-        done < <(gluster volume list)
+        done < <(block_volnames_brick_count_get)
         if [[ ! -z "$wait_info" ]]; then echo "$wait_info"; fi
 }
 
 function check_glusterd_running()
 {
-        if ! pidof glusterd > /dev/null 2>&1
+        if ! command -v ps > /dev/null 2>&1
+        then
+                printLog "ERROR: 'ps' command is not found"
+                exit 1
+        fi
+
+# Explanation of why ps command is used this way:
+# https://stackoverflow.com/questions/9117507/linux-unix-command-to-determine-if-process-is-running
+        if ! ps cax | grep -w '[g]lusterd' > /dev/null 2>&1
```

Comment:
instead of redirecting to `> /dev/null`, use `grep -q`

Reply:
Cool, will do

Reply:
Found this in manpage, so won't change this part:        Normally the exit status is 0 if a line is selected, 1 if no lines were selected, and
       2 if an error occurred.  However, if the -q or --quiet or --silent is used and a line
       is selected, the exit status is 0 even if an error occurred.